orchestration/dap_orchestration/resources.py

```diff
@@ -0,0 +1,135 @@
+import subprocess
+from typing import Any, List
+from dataclasses import dataclass
+
+from dagster import DagsterLogManager, Array, Field, resource, StringSource, IntSource
+from dagster.core.execution.context.init import InitResourceContext
+
+
+@dataclass
+class DataflowBeamRunner:
+    working_dir: str
+    logger: DagsterLogManager
+    # todo: **kwargs param to local beam runner + dataflow beam runner when we refactor this to use dagster_utils
+    region: str
+    worker_machine_type: str
+    autoscaling_algorithm: str
+    num_workers: int
+    max_num_workers: int
+    google_project: str
+
+    # note: Use "-> None" if function does not return a value
+    def __post_init__(self) -> None:
+        self.arg_dict = {
+            # resource config - how stuff behaves
+            # will not affect what is returned - will only affect run performance/success
+            "region": self.region,
+            "workerMachineType": self.worker_machine_type,
+            "autoscalingAlgorithm": self.autoscaling_algorithm,
+            "numWorkers": str(self.num_workers),
+            "maxNumWorkers": str(self.max_num_workers),
+            "project": self.google_project,
+            # hardcode anything that doesn't fall into solids/resources ie. dataflow
+            "runner": "dataflow",
+            # To use the service-based Dataflow Shuffle in your batch pipelines
+            "experiments": "shuffle_mode=service",
+        }
+
+    # pull the target_class out of the resource config
+
+    def run(
+            self,
+            run_arg_dict: dict[str, Any],
+    ) -> None:
+        # create a new dictionary containing the keys and values of arg_dict + solid arguments
+        dataflow_run_flags = {**self.arg_dict, **run_arg_dict}
+        self.logger.info("Dataflow beam runner")
+        # pull target_class out of list of flags
```

Comment:
⛏️ I think this comment as-is is of similarly little value; I _do_ feel one is warranted for these special cases though. Perhaps:

```
# special case target_class and scala_project as those cannot be placed in the args list of the sbt invocation`?